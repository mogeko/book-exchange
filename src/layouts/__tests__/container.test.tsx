import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import Container from "@/layouts/container";

describe("Container", () => {
  it("should render correctly", () => {
    render(
      <Container>
        <div>hello, world!</div>
      </Container>
    );
    expect(screen.getAllByText(/hello, world!/)[0]).toBeInTheDocument();
  });

  it("snapshot a container", () => {
    const { container } = render(
      <Container>
        <div>hello, world!</div>
      </Container>
    );
    expect(container).toMatchSnapshot();
  });
});
