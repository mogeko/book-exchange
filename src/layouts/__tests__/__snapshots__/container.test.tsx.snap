// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container snapshot a container 1`] = `
<div>
  <div
    class="flex flex-col min-h-screen min-w-full w-full"
    id="container"
  >
    <div>
      hello, world!
    </div>
  </div>
</div>
`;

exports[`Container snapshot a layout with Header and Footer 1`] = `
<div>
  <div
    class="drawer"
  >
    <input
      class="drawer-toggle"
      id="mobile-menu"
      type="checkbox"
    />
    <div
      class="drawer-content"
    >
      <div
        class="flex flex-col min-h-screen min-w-full w-full"
        id="container"
      >
        <div
          class="min-w-full"
        >
          <div>
            Error...
          </div>
          <div
            class="bg-inherit p-2 md:p-0 md:bg-base-300"
          >
            <header
              class="navbar p-0 m-auto rounded-md max-w-screen-lg bg-base-300 md:rounded-none"
            >
              <div
                class="navbar-start sm:hidden"
              >
                <label
                  class="btn btn-ghost btn-circle"
                  for="mobile-menu"
                >
                  <svg
                    aria-hidden="true"
                    class="h-6 w-6"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 20 20"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M3 5a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 10a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 15a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </label>
              </div>
              <div
                class="navbar-center sm:navbar-start"
              >
                <div
                  class="flex justify-start"
                >
                  <a
                    class="btn gap-1 btn-ghost normal-case"
                    href="/"
                  >
                    <span
                      class="sr-only"
                    >
                      Bookworm
                    </span>
                    <span
                      style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                    >
                      <span
                        style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                      >
                        <img
                          alt=""
                          aria-hidden="true"
                          src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
                          style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                        />
                      </span>
                      <img
                        alt="logo"
                        data-nimg="intrinsic"
                        decoding="async"
                        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                      />
                      <noscript />
                    </span>
                    <h1
                      class="text-xl font-bold text-base-content"
                    >
                      Bookworm
                    </h1>
                  </a>
                </div>
              </div>
              <div
                class="navbar-end"
              >
                <div
                  class="hidden md:form-control"
                >
                  <input
                    class="input input-bordered input-primary input-sm w-96"
                    placeholder="Search"
                    type="text"
                  />
                </div>
                <a
                  class="btn btn-ghost btn-circle md:hidden"
                  href="/#"
                >
                  <svg
                    aria-hidden="true"
                    class="h-6 w-6"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 20 20"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M8 4a4 4 0 100 8 4 4 0 000-8zM2 8a6 6 0 1110.89 3.476l4.817 4.817a1 1 0 01-1.414 1.414l-4.816-4.816A6 6 0 012 8z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </a>
                <div
                  class="dropdown dropdown-end sm:hidden"
                >
                  <label
                    class="btn btn-ghost btn-circle avatar"
                    tabindex="0"
                  >
                    <div
                      class="w-6 rounded-full"
                    >
                      <svg
                        aria-hidden="true"
                        class="h-6 w-6"
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        viewBox="0 0 20 20"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          clip-rule="evenodd"
                          d="M10 9a3 3 0 100-6 3 3 0 000 6zm-7 9a7 7 0 1114 0H3z"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </div>
                  </label>
                  <ul
                    class="menu menu-compact dropdown-content mt-3 p-2 shadow bg-base-200 rounded-box w-52"
                    tabindex="0"
                  >
                    <li>
                      <a
                        class="justify-between"
                        href="/#"
                      >
                        Profile
                        <span
                          class="badge"
                        >
                          New
                        </span>
                      </a>
                    </li>
                    <li>
                      <a
                        class="justify-between"
                        href="/#"
                      >
                        Settings
                      </a>
                    </li>
                    <li>
                      <a
                        class="justify-between"
                        href="/#"
                      >
                        Logout
                      </a>
                    </li>
                  </ul>
                </div>
              </div>
            </header>
          </div>
        </div>
        <main
          class="flex-1 max-w-screen-lg w-full mx-auto p-6 md:p-4"
        >
          <div>
            hello, world!
          </div>
        </main>
        <footer
          class="footer footer-center p-4 bg-base-300 text-base-content"
        >
          <div>
            <p>
              Copyright Â© 
              2022
              , All right reserved
               by Zheng Junyi
              .
            </p>
          </div>
        </footer>
      </div>
    </div>
    <div>
      Error...
    </div>
  </div>
</div>
`;
